diff --git a/out/package.js b/out/package.js
index 5bc5e0c284a7fe533d55886693fb360f85b9a1a3..a446da7994a99b07d0e150d9a3d31ba622f50ab5 100644
--- a/out/package.js
+++ b/out/package.js
@@ -1136,7 +1136,7 @@ async function collectAllFiles(cwd, dependencies, dependencyEntryPoints) {
 }
 function collectFiles(cwd, dependencies, dependencyEntryPoints, ignoreFile) {
     return collectAllFiles(cwd, dependencies, dependencyEntryPoints).then(files => {
-        files = files.filter(f => !/\r$/m.test(f));
+        files = files.filter((f, idx, arr) => !/\r$/m.test(f) && arr.indexOf(f) === idx);
         return (fs.promises
             .readFile(ignoreFile ? ignoreFile : path.join(cwd, '.vscodeignore'), 'utf8')
             .catch(err => err.code !== 'ENOENT' ? Promise.reject(err) : ignoreFile ? Promise.reject(err) : Promise.resolve(''))
@@ -1341,6 +1341,9 @@ exports.listFiles = listFiles;
  * Lists the files included in the extension's package. Runs prepublish.
  */
 async function ls(options = {}) {
+    const cwd = options.cwd || process.cwd();
+    const manifest = await readManifest(cwd);
+    util.patchOptionsWithManifest(options, manifest);
     const files = await listFiles({ ...options, prepublish: true });
     for (const file of files) {
         console.log(`${file}`);
